File: BOJ/python3/2748.py
'''
피보나치 수 2
https://www.acmicpc.net/problem/2748
'''
import sys
import time
start = time.time()  # 시작 시간
sys.stdin = open("input.txt", "r")  # input.txt 파일로 입력


def fibo(n):
    if n <= 1:
        dp[n] = n
        return dp[n]
    elif dp[n] != 0:
        return dp[n]

    dp[n] = fibo(n-1) + fibo(n-2)
    return dp[n]


n = int(input())

dp = [0] * (n+1)
ans = fibo(n)

print(ans)


# 현재시각 - 시작시간 = 실행 시간(sec)
print("--------------------")
print(f"Elapsed Time: {round(time.time() - start, 3)}s")

File: BOJ/nodejs/2748.js
/*
피보나치 수 2
https://www.acmicpc.net/problem/2748
 */
const start = new Date().getTime();
const fs = require("fs");
const input = fs.readFileSync("input.txt").toString().split("\n");
//const input = fs.readFileSync("/dev/stdin").toString().split("\n");

let n = 1 * input[0];
const fibo = Array.from({ length: n + 1 }, () => -1);

if (n >= 1) {
  fibo[0] = 0n; // bigint
  fibo[1] = 1n;
}
for (let i = 2; i < n + 1; i++) {
  fibo[i] = fibo[i - 1] + fibo[i - 2];
}

console.log(fibo[n].toString());

const end = new Date().getTime();
console.log("--------------------");
console.log(`Elapsed Time: ${(new Date().getTime() - start) / 1000}s`);
